### PR TITLE
Add's support for Juniper Contrail packaging + test-kitchen

### PR DESCRIPTION
**Pull request by epcim**
TODO:
- some sutes not yet passing

Questions:
- mos8/9 dependency should not be required - but close tested stable.

### CODE REVIEW

File: opencontrail/files/2.2/config.global.js
Comment (fpytloun) [inline opencontrail/files/2.2/config.global.js]: Strip whitespaces by using `-%}`

File: opencontrail/files/3.0/contrail-vrouter-agent.conf
Comment (fpytloun) [inline opencontrail/files/3.0/contrail-vrouter-agent.conf]: Strip whitespace or you can use `import compute, common` on one line

**Review by pupapaik — COMMENTED** [review]
Do not merge it before my approval! I need to review.

**Comment by epcim** [comment]
This is WIP. Obviously not ready yet.
Will update title later.

File: opencontrail/files/2.2/config.global.js
Comment (epcim) [reply opencontrail/files/2.2/config.global.js]: %- at start of block i guess is ok

File: opencontrail/map.jinja
Comment (epcim) [inline opencontrail/map.jinja]: contrail-vrouter-init is missing in latest formula, was it removed, any idea if needed with juniper packaging?

File: .kitchen.yml
Comment (epcim) [inline .kitchen.yml:108]: We have two options
- run travis not with docker but in VM using kitchen-local etc.. 
- wrap conflicting states (kernel modules insmod) to be skipped during CI

File: .kitchen.yml
Comment (epcim) [inline .kitchen.yml:115]: package opencontrail-kubelet is missing in public repositories to pass this suite

File: README.rst
Comment (LotharKAtt) [inline README.rst]: You mean vendor?

File: opencontrail/map.jinja
Comment (LotharKAtt) [inline opencontrail/map.jinja]: 'contrail-database' correct name of pkg is **contrail-openstack-database**

**Review by LotharKAtt — CHANGES_REQUESTED** [review]

File: opencontrail/map.jinja
Comment (epcim) [reply opencontrail/map.jinja]: also removed contrail-vrouter (virtual pkg, as packages from apt-mk repo complains, to install just the specific one - not the virtual pkg) CC @pupapaik  - pls have a look

File: opencontrail/map.jinja
Comment (epcim) [inline opencontrail/map.jinja]: @pupapaik - see the other comments related to removal of contrail-vrouter virtual package.

File: README.rst
Comment (pupapaik) [inline README.rst]: I would not reference specific versions of Juniper, because then we have to update readme even if content is same. It will work with all 3.0.2.x versions.

File: opencontrail/config.sls
Comment (pupapaik) [inline opencontrail/config.sls:158]: This is obsolete. So do not add this into condition.

File: opencontrail/control.sls
Comment (pupapaik) [inline opencontrail/control.sls]: Ifmap is not related to control, but config role! Please remove this from control.sls.

File: opencontrail/config.sls
Comment (pupapaik) [inline opencontrail/config.sls:160]: Put condition here, if juniper contrail then do not absent this file. Or maybe we can move ifmap for opencontrail under supervisor as well.

File: opencontrail/files/2.2/contrail-tor-agent.conf
Comment (pupapaik) [inline opencontrail/files/2.2/contrail-tor-agent.conf]: Do you really use tor agents and why 2.2 release?

File: opencontrail/files/2.2/config.global.js
Comment (pupapaik) [inline opencontrail/files/2.2/config.global.js]: Please skip 2.2 release for Juniper Contrail. It is obsolete.

File: opencontrail/files/2.2/contrail-vrouter-agent.conf
Comment (pupapaik) [inline opencontrail/files/2.2/contrail-vrouter-agent.conf]: Same here. Do not touch 2.2 release files please.

File: opencontrail/files/3.0/config.global.js
Comment (pupapaik) [inline opencontrail/files/3.0/config.global.js:189]: is this valid? This is not needed, right? We can use same for both.

File: opencontrail/files/ifmap.ini
Comment (pupapaik) [inline opencontrail/files/ifmap.ini]: This file is not needed. It comes from package.

File: opencontrail/map.jinja
Comment (pupapaik) [reply opencontrail/map.jinja]: I have pull request where I leave only contrail-vrouter-agent, because dkms or dpdk depends on target.

PM: have to retest. Juniper pkg's complains to choose specific version of contrail-vrouter -???? instead of virtual package. Your PR will probably will not pass CI in docker environment.

**Review by pupapaik — DISMISSED** [review]
Please see my comments.

File: opencontrail/files/2.2/contrail-tor-agent.conf
Comment (epcim) [reply opencontrail/files/2.2/contrail-tor-agent.conf]: this is fix for CI to pass, system.name is populated by reclass service which is not available if you don't use reclass, but pure pillars.

File: opencontrail/files/2.2/config.global.js
Comment (epcim) [reply opencontrail/files/2.2/config.global.js]: Readme has info of supported version 3.0.x. Do you really want to remove any change under files/2.2?

File: opencontrail/files/3.0/config.global.js
Comment (epcim) [reply opencontrail/files/3.0/config.global.js:189]: it's the location where it's created by package if I do understand.  We were forced to fix the path in order to make it runing. is the redis dump file created by formula, I don't think so..

File: opencontrail/files/ifmap.ini
Comment (epcim) [reply opencontrail/files/ifmap.ini]: Hmm.. our opencontrail formula remove's it. This was assumed aka fix that state. If the file not exist, you have to reinstall while recreate missing files (common mistake comes here). I will update condition not to render this template if the destination already exist.

File: opencontrail/config.sls
Comment (epcim) [reply opencontrail/config.sls:158]: this is not obsolate, as we really don't want to remove that file for juniper packaging..

**Comment by epcim** [comment]
Ready for merge, please update review.

**Comment by pupapaik** [comment]
This week I will test it. wait for approval.

**Comment by epcim** [comment]
Issue found, On fresh NAL nodes with juniper packagin, was required to create:

chown -R cassandra.cassandra /var/lib/cassandra
mkdir /var/lib/cassandra

**Comment by epcim** [comment]
One another shit with juniper repos is that for NAL//NTW before you may run states, you need to downgrade gcc-4.8-base libc6 libstdc++6 tzdata

todo allow downgrades -y

root@ntw-03:/var/log# apt install contrail-openstack-database
...
The following packages will be DOWNGRADED:
gcc-4.8-base libc6 libstdc++6 tzdata
1 upgraded, 98 newly installed, 4 downgraded, 0 to remove and 88 not upgraded.
Need to get 120 MB of archives.
After this operation, 239 MB of additional disk space will be used.
Do you want to continue? [Y/n] y^C

File: opencontrail/collector.sls
Comment (pupapaik) [inline opencontrail/collector.sls]: put it there without condition. It is not needed.

File: opencontrail/config.sls
Comment (pupapaik) [inline opencontrail/config.sls]: put it there without condition. It is not needed.

File: opencontrail/control.sls
Comment (pupapaik) [inline opencontrail/control.sls]: put it there without condition. It is not needed.

File: opencontrail/database.sls
Comment (pupapaik) [inline opencontrail/database.sls]: put it there without condition. It is not needed.

File: opencontrail/database.sls
Comment (pupapaik) [inline opencontrail/database.sls:140]: you do not need to do this, if  you created data directory in previsous step.

File: opencontrail/files/3.0/contrail-vrouter-agent.conf
Comment (pupapaik) [inline opencontrail/files/3.0/contrail-vrouter-agent.conf:1]: why you import common here? It is not needed.

File: opencontrail/web.sls
Comment (pupapaik) [inline opencontrail/web.sls]: put it there without condition. It is not needed.

**Review by pupapaik — CHANGES_REQUESTED** [review]
Please check my comments

File: opencontrail/database.sls
Comment (epcim) [reply opencontrail/database.sls:140]: but not all directories/files are re-owned

Comment (pupapaik) [reply opencontrail/database.sls:140]: Please remove it.

**Pull request merged**
